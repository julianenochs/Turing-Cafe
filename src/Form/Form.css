.form__section {
    display: flex;
    font-size: 1.5em;
}

.form__element {
    display: flex;
    justify-content: space-evenly;
}

button {
    font-size: 1em;
}