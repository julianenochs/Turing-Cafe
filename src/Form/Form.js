import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
    constructor() {
        super();
        this.state = {

        }
    }

    componentDidMount() {

    }

    render() {
        return(
            <section className='form__section'>
                <input 
                    type='text'
                    name='name'
                    // value={}
                    placeholder='Name'/>
                <input
                    type='date'
                    name='name'
                    // value={}
                    placeholder='Date' />
                <input
                    type='time'
                    name='name'
                    // value={}
                    placeholder='Time' />
                <input
                    type='text'
                    name='name'
                    // value={}
                    placeholder='Number of Guests' />
                <button>Make Reservation</button>
            </section>
        )
    }
}

export default Form;