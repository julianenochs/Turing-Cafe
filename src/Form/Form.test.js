import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './Form';

describe('Form', () => {
    let wrapper
    let mockAddReservation = jest.fn();
    let mockName = 'Julian'
    let mockDate = '2019/10/13'
    let mockTime = '11:00pm'
    let mockNumber = 3
    beforeEach(() => {
        wrapper = shallow(<Form 
                    name={mockName}
                    date={mockDate}
                    time={mockTime}
                    number={mockNumber}
                    addReservation={mockAddReservation} />)
    });
    
    it('should match the snapshot with all of the data passed through', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call addReservation when the button is clicked', () => {
        wrapper = mount(<Form
            name={mockName}
            date={mockDate}
            time={mockTime}
            number={mockNumber}
            addReservation={mockAddReservation} />)
        wrapper.find('button').simulate('click');
        expect(wrapper.props().addReservation).toHaveBeenCalled();
    });

    it('should reset the state to default settings when clearInputs is called', () => {
        wrapper.instance().clearInputs
        expect(wrapper.state().name).toEqual('');
        expect(wrapper.state().date).toEqual('');
        expect(wrapper.state().time).toEqual('');
        expect(wrapper.state().number).toEqual(0);
    });
});