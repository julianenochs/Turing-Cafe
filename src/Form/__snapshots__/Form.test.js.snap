// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all of the data passed through 1`] = `
<section
  className="form__section"
>
  <input
    className="form__element"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form__element"
    name="date"
    onChange={[Function]}
    placeholder="Date"
    type="date"
    value=""
  />
  <input
    className="form__element"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="time"
    value=""
  />
  <input
    className="form__element"
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
    value={0}
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</section>
`;
